test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,207 @@
+	flag.BoolVar(&skipCleanup, "e2e.skip-resource-cleanup", false, "if true, the resource cleanup after tests will be skipped")
+}
+
+func TestE2e(t *testing.T) {
+	g := NewWithT(t)
+	ctrl.SetLogger(klog.Background())
+
+	// ensure the artifacts folder exists
+	g.Expect(os.MkdirAll(artifactFolder, 0755)).To(Succeed(), "Invalid test suite argument. Can't create e2e.artifacts-folder %q", artifactFolder) //nolint:gosec
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2e Suite")
+}
+
+var _ = SynchronizedBeforeSuite(func() []byte {
+	var kubeconfigPath string
+	Expect(configPath).To(BeAnExistingFile(), "Invalid test suite argument. e2e.config should be an existing file.")
+	e2eConfig = LoadE2EConfig(configPath)
+
+	if useExistingCluster {
+		kubeconfigPath = os.Getenv("KUBECONFIG")
+		if kubeconfigPath == "" {
+			kubeconfigPath = os.Getenv("HOME") + "/.kube/config"
+		}
+	} else {
+		clusterProvider = bootstrap.CreateKindBootstrapClusterAndLoadImages(ctx, bootstrap.CreateKindBootstrapClusterAndLoadImagesInput{
+			Name:   "bmo-e2e",
+			Images: e2eConfig.Images,
+		})
+		Expect(clusterProvider).ToNot(BeNil(), "Failed to create a cluster")
+		kubeconfigPath = clusterProvider.GetKubeconfigPath()
+	}
+	Expect(kubeconfigPath).To(BeAnExistingFile(), "Failed to get the kubeconfig file for the cluster")
+
+	scheme := runtime.NewScheme()
+	framework.TryAddDefaultSchemes(scheme)
+	clusterProxy := framework.NewClusterProxy("bmo-e2e", kubeconfigPath, scheme)
+	Expect(clusterProxy).ToNot(BeNil(), "Failed to get a cluster proxy")
+
+	DeferCleanup(func() {
+		clusterProxy.Dispose(ctx)
+	})
+
+	os.Setenv("KUBECONFIG", clusterProxy.GetKubeconfigPath())
+
+	if e2eConfig.GetVariable("DEPLOY_CERT_MANAGER") != "false" {
+		// Install cert-manager
+		By("Installing cert-manager")
+		cmd := exec.Command("cmctl", "x", "install")
+		output, err := cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("%s", output))
+		By("Checking that the cert-manager API is available")
+		cmd = exec.Command("cmctl", "check", "api")
+		output, err = cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("%s", output))
+	}
+
+	if e2eConfig.GetVariable("DEPLOY_BMO") != "false" {
+		// Install BMO
+		By("Installing BMO")
+		kustomization := e2eConfig.GetVariable("BMO_KUSTOMIZATION")
+		cmd := exec.Command("kubectl", "--kubeconfig", clusterProxy.GetKubeconfigPath(), "apply", "-k", kustomization) //nolint:gosec
+		output, err := cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("%s", output))
+
+		bmoDeployment := &v1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "baremetal-operator-controller-manager",
+				Namespace: "baremetal-operator-system",
+			},
+		}
+		// Wait for it to become available
+		framework.WaitForDeploymentsAvailable(ctx, framework.WaitForDeploymentsAvailableInput{
+			Getter:     clusterProxy.GetClient(),
+			Deployment: bmoDeployment,
+		}, "5m")
```

Comment:
I think we can add all intervals to the config file

Reply:
Oh good catch!